backend/danswer/search/retrieval/search_runner.py

```diff
@@ -163,7 +174,62 @@ def doc_index_retrieval(
         else:
             raise RuntimeError("Invalid Search Flow")
 
-    return cleanup_chunks(top_chunks)
+    retrieval_requests: list[VespaChunkRequest] = []
```

Comment:
This part of the code removes the mega chunks, saves the score of each mega chunk, retrieves the original chunks referenced by each mega chunk, gives the score of the mega chunk to the normal chunks, dedupes the normal chunks (keeping the higher score) and finally reorders with descending score